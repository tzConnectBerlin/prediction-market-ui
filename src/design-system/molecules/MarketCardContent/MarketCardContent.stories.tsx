import { Story, Meta } from '@storybook/react/types-6-0';
import { MarketCardContent, MarketCardContentProps } from './MarketCardContent';
import { Currency, TokenType } from '../../../interfaces/market';

export default {
  title: 'Molecule/MarketCardContent',
  component: MarketCardContent,
} as Meta;

const Template: Story<MarketCardContentProps> = (args) => <MarketCardContent {...args} />;

export const TokenAndStatisticList = Template.bind({});
TokenAndStatisticList.args = {
  tokenList: [
    {
      type: TokenType.yes,
      value: 14,
    },
    {
      type: TokenType.no,
      value: 5,
    },
  ],
  statisticList: [
    {
      type: 'WEEKLY',
      value: '95',
      changes: 'up',
      currency: Currency.USD,
      tokenType: TokenType.yes,
    },
    {
      type: 'VOLUME',
      value: '5',
      currency: Currency.USD,
    },
  ],
};
