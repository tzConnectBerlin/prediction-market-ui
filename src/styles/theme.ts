import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
