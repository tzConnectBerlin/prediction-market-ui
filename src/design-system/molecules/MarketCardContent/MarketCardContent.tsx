import * as React from 'react';
import { Grid } from '@mui/material';
import styled from '@emotion/styled';
import { lightTheme as theme } from '../../../styles/theme';
import { MarketCardStatistic, MarketCardToken, TokenType } from '../../../interfaces/market';
import { Typography } from '../../atoms/Typography';

interface StyledLabelProps {
  icon?: string;
  fontColor?: string;
}

const StyledGrid = styled(Grid)`
  font-size: 0.8em;
  padding: 1em;
  padding-top: 0;
  margin-top: auto;
`;

const StyledLabel = styled.div<StyledLabelProps>`
  color: ${({ fontColor }) => fontColor};
  padding: 0.2em;
  &.hasIcon {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    & > * {
      margin-left: 0.2em;
    }
  }
`;
export interface MarketCardContentProps {
  /**
   * market token list
   */
  tokenList?: MarketCardToken[];
  /**
   * market statistic List
   */
  statisticList?: MarketCardStatistic[];
}

const defaultTokenList: MarketCardToken[] = [];
const defaultStatisticList: MarketCardStatistic[] = [];

export const MarketCardContent: React.FC<MarketCardContentProps> = ({
  tokenList = defaultTokenList,
  statisticList = defaultStatisticList,
}) => {
  const getTokenList = () => {
    return tokenList.map((token, i) => {
      const color =
        token.type === TokenType.yes ? theme.palette.success.main : theme.palette.error.main;
      return (
        <Grid item xs={6} key={i}>
          <StyledLabel fontColor={theme.palette.text.secondary}>
            <Typography size="h4" component="h3">
              {token.type}
            </Typography>
          </StyledLabel>
          <StyledLabel fontColor={color}>
            <Typography size="h3">{token.value}%</Typography>
          </StyledLabel>
        </Grid>
      );
    });
  };

  const getStatisticList = () => {
    return statisticList.map((item, i) => {
      const color =
        item.tokenType === TokenType.yes
          ? theme.palette.success.main
          : item.tokenType === TokenType.no
          ? theme.palette.error.main
          : undefined;

      return (
        <Grid item xs={6} key={i}>
          <StyledLabel
            fontColor={theme.palette.text.secondary}
            className={item.changes ? 'hasIcon' : ''}
          >
            <Typography size="h4" component="h3">
              {item.type}
            </Typography>
          </StyledLabel>
          <StyledLabel fontColor={color}>
            <Typography size="h3">
              {item.tokenType} {item.value}
            </Typography>
          </StyledLabel>
        </Grid>
      );
    });
  };

  return (
    <StyledGrid container spacing={1}>
      <Grid container item xs={12} spacing={3}>
        {tokenList.length > 0 && getTokenList()}
      </Grid>
      <Grid container item xs={12} spacing={3}>
        {statisticList.length > 0 && getStatisticList()}
      </Grid>
    </StyledGrid>
  );
};
