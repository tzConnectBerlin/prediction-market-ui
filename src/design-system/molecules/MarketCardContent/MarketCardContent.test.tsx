import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { lightTheme as theme } from '../../../theme';
import { MarketCardContent, MarketCardContentProps } from './MarketCardContent';

const defaultProps: MarketCardContentProps = {
  tokenList: [
    {
      label: 'YES',
      value: '95$',
      valueColor: theme.palette.success.main,
    },
    {
      label: 'NO',
      value: '5$',
      valueColor: theme.palette.error.main,
    },
  ],
  statisticList: [
    {
      label: 'WEEKLY',
      value: 'YES 95$',
      valueColor: theme.palette.success.main,
      changes: 'up',
    },
    {
      label: 'VOLUME',
      value: '5$',
      valueColor: theme.palette.text.primary,
    },
  ],
};

describe('Snapshot testing MarketCardContent Component', () => {
  it('renders correctly without Props', () => {
    const cardContent = renderer.create(<MarketCardContent />).toJSON();
    expect(cardContent).toMatchSnapshot();
  });

  it('renders correctly only tokenList', () => {
    const cardContent = renderer
      .create(<MarketCardContent tokenList={defaultProps.tokenList} />)
      .toJSON();
    expect(cardContent).toMatchSnapshot();
  });

  it('renders correctly only statisticList', () => {
    const cardContent = renderer
      .create(<MarketCardContent statisticList={defaultProps.statisticList} />)
      .toJSON();
    expect(cardContent).toMatchSnapshot();
  });

  it('renders correctly with Props', () => {
    const cardContent = renderer
      .create(
        <MarketCardContent
          tokenList={defaultProps.tokenList}
          statisticList={defaultProps.statisticList}
        />,
      )
      .toJSON();
    expect(cardContent).toMatchSnapshot();
  });
});

describe('Element testing MarketCardContent Component', () => {
  it('render correctly statisticList', async () => {
    const { getByText, container } = render(
      <MarketCardContent statisticList={defaultProps.statisticList} />,
    );

    expect(getByText(/WEEKLY/i)).toBeInTheDocument();
    expect(getByText(/VOLUME/i)).toBeInTheDocument();
    expect(container.querySelector('svg')).toBeInTheDocument();
  });

  it('render correctly tokenList', async () => {
    const { getByText, container } = render(
      <MarketCardContent tokenList={defaultProps.tokenList} />,
    );

    expect(getByText(/YES/i)).toBeInTheDocument();
    expect(container.querySelector('svg')).not.toBeInTheDocument();
  });
});
