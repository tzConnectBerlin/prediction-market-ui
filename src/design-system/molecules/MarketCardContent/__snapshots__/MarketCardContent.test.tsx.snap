// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketCardContent Component renders correctly only statisticList 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-0>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 24px);
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-flex-basis: calc(100% + 24px);
  -ms-flex-preferred-size: calc(100% + 24px);
  flex-basis: calc(100% + 24px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 24px);
}

.emotion-1>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

.emotion-4 {
  color: rgba(29, 34, 39, 0.65);
  padding: 0.2em;
}

.emotion-4.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-5.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-6 {
  color: rgba(42, 184, 0, 1);
  padding: 0.2em;
}

.emotion-6.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-7.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-11 {
  padding: 0.2em;
}

.emotion-11.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11.hasIcon>* {
  margin-left: 0.2em;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-3"
    >
      <div
        className="hasIcon emotion-4"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-5"
          lines={3}
        >
          WEEKLY
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
            />
          </svg>
        </h4>
      </div>
      <div
        className="emotion-6"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-7"
          lines={3}
        >
          Yes
           
          95
        </h3>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-3"
    >
      <div
        className=" emotion-4"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-5"
          lines={3}
        >
          VOLUME
           
        </h4>
      </div>
      <div
        className="emotion-11"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-7"
          lines={3}
        >
           
          5
        </h3>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing MarketCardContent Component renders correctly only tokenList 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-0>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 24px);
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-flex-basis: calc(100% + 24px);
  -ms-flex-preferred-size: calc(100% + 24px);
  flex-basis: calc(100% + 24px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 24px);
}

.emotion-1>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

.emotion-3 {
  color: rgba(29, 34, 39, 0.65);
  padding: 0.2em;
}

.emotion-3.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5 {
  color: rgba(42, 184, 0, 1);
  padding: 0.2em;
}

.emotion-5.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-6.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  color: rgba(255, 57, 57, 1);
  padding: 0.2em;
}

.emotion-10.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10.hasIcon>* {
  margin-left: 0.2em;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className="emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          Yes
        </h4>
      </div>
      <div
        className="emotion-5"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
          14
          %
        </h3>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className="emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          No
        </h4>
      </div>
      <div
        className="emotion-10"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
          5
          %
        </h3>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  />
</div>
`;

exports[`Snapshot testing MarketCardContent Component renders correctly with Props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-0>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 24px);
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-flex-basis: calc(100% + 24px);
  -ms-flex-preferred-size: calc(100% + 24px);
  flex-basis: calc(100% + 24px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 24px);
}

.emotion-1>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
}

.emotion-3 {
  color: rgba(29, 34, 39, 0.65);
  padding: 0.2em;
}

.emotion-3.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5 {
  color: rgba(42, 184, 0, 1);
  padding: 0.2em;
}

.emotion-5.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
}

.emotion-6.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  color: rgba(255, 57, 57, 1);
  padding: 0.2em;
}

.emotion-10.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10.hasIcon>* {
  margin-left: 0.2em;
}

.emotion-21 {
  padding: 0.2em;
}

.emotion-21.hasIcon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-21.hasIcon>* {
  margin-left: 0.2em;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className="emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          Yes
        </h4>
      </div>
      <div
        className="emotion-5"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
          14
          %
        </h3>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className="emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          No
        </h4>
      </div>
      <div
        className="emotion-10"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
          5
          %
        </h3>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className="hasIcon emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          WEEKLY
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
            />
          </svg>
        </h4>
      </div>
      <div
        className="emotion-5"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
          Yes
           
          95
        </h3>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 emotion-2"
    >
      <div
        className=" emotion-3"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4  emotion-4"
          lines={3}
        >
          VOLUME
           
        </h4>
      </div>
      <div
        className="emotion-21"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3  emotion-6"
          lines={3}
        >
           
          5
        </h3>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing MarketCardContent Component renders correctly without Props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 8px);
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  font-size: 0.8em;
  padding: 1em;
  margin-top: auto;
}

.emotion-0>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-0>.MuiGrid-item {
  padding-left: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: calc(100% + 24px);
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  -webkit-flex-basis: calc(100% + 24px);
  -ms-flex-preferred-size: calc(100% + 24px);
  flex-basis: calc(100% + 24px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 24px);
}

.emotion-1>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 24px;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-grid-xs-12 emotion-1"
  />
</div>
`;
